feat(pie-docs): DSW-2126 add textarea overview page (#1670)
justeattakeaway/pie: apps/pie-docs/src/components/textarea/overview/overview.md

        ]
    }
} %}

---

## Anatomy

{% contentPageImage {
    src:"../../../assets/img/components/textarea/anatomy.svg",
    alt: "Textarea with numbers specifying four different parts of textarea anatomy.",
    width: "416px"
} %}

{% list {
    type: listTypes.ordered,
    items: [
        "**Form label (optional):** The [form label](/components/form-label) provides necessary content / information to the form.",
        "**String:** Placeholder or user inputted value.",
        "**Resize tab (optional):** Allows the user to manually resize the form container.",
        "**Assistive text (optional):** The [assistive text](/components/assistive-text/) provides additional instructional information / error / success messaging."
    ]
} %}

---

## Variations

### Default

{% contentPageImage {
    src:"../../../assets/img/components/textarea/default.svg",
    alt: "Textarea component with a label to specify a default state.",
    width: "256px"
} %}

---

## Modifiers

### Type

{% contentLayout %}
  {% contentItem %}
    <h4>Placeholder</h4>

>     <p>Placeholder text should not contain crucial information, and only displayed if beneficial to the user.</p>

Reviewer: ```suggestion
    <p>Placeholder text should not contain crucial information, and should only be displayed if beneficial to the user.</p>
```

    {% contentPageImage {
      src: "../../../assets/img/components/textarea/default.svg",
      width: "256px",
      alt: "A blank textarea with placeholder followed by the label text."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Filled</h4>
    <p>Placeholder text within the input field disappears after the user begins entering data into the input.</p>
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/filled.svg",
      width: "256px",
      alt: "Textarea component with a label and string content inside it."
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

### Height

The height, unlike the width, of a textarea is flexible, with two different options available.

{% contentLayout %}
  {% contentItem %}
    <h4>Auto-resize</h4>
    <p>Used when the intended behaviour is for the form's height to dynamically adjust based on input length.</p>
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/default.svg",
      width: "256px",
      alt: "Textarea component with auto-resizing option."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Manual-resize</h4>

>     <p>Used when the intended behavior is for the user to have control over the form’s height.</p>

Reviewer: Should be consistent with description for auto resize
```suggestion
    <p>Used when the intended behaviour is for the user to have control over the form’s height.</p>
```

    {% contentPageImage {
      src: "../../../assets/img/components/textarea/manual-resize.svg",
      width: "256px",
      alt: "Textarea component with manual-resizing option."
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

---

## Sizes

Use a consistent height when it is being used alongside other form components on the same page.

{% contentLayout %}
  {% contentItem %}
    <h4>Small</h4>
    <p>Use when there isn’t enough vertical space for the default textarea size.</p>
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/small.svg",
      width: "206px",
      alt: "Textarea component with a small height."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Medium</h4>

>     <p>Default size and should be used whenever possible.</p>

Author: ```suggestion
    <p>Default size and should always be used wherever it is practically possible.</p>
```

    {% contentPageImage {
      src: "../../../assets/img/components/textarea/medium.svg",
      width: "206px",
      alt: "Textarea component with a medium height."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Large</h4>
    <p>Choose this size when there is a lot of space to work with. This size is typically used in simple forms or when a textarea is placed by itself on a page.</p>
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/large.svg",
      width: "206px",
      alt: "Textarea component with a large height."
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

#### Auto-resize

{% componentDetailsTable {
  tableData: autoResize
} %}

#### Manual-resize

{% componentDetailsTable {
  tableData: manualResize
} %}

---

## Content

### Placeholder

> Use clear placeholder text for the text input trigger so that users understand the purpose.

Reviewer: ```suggestion
Use clear placeholder text so that users understand the purpose of the textarea.
```

{% contentPageImage {
    src:"../../../assets/img/components/textarea/placeholder.svg",
    alt: "Textarea component with a placeholder.",
    width: "256px"
} %}

### Input types

#### Alphanumeric

> Both letters and numbers are allowed, for all use-cases this will be used.

Reviewer: For the text input we also mentioned special characters. Omitting this makes it sound to me like they are not allowed.

Author: ```suggestion
Both letters, numbers and certain special characters are allowed, for the majority of use-cases this will be used.
```

} %}

{% contentLayout %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/manual-resize-focused-small.svg",
      width: "256px",
      alt: "A small-sized textarea component with manual-resize functionality is focused and has text inside it."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/manual-resize-focused-medium.svg",
      width: "256px",
      alt: "A medium-sized textarea component with manual-resize functionality is focused and has text inside it."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/manual-resize-focused-large.svg",
      width: "256px",
      alt: "A large-sized textarea component with manual-resize functionality is focused and has text inside it."
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

### Character counter

The character limit and counter is used to let users know how long their entry can be before they begin typing.

{% notification {
  type: "warning",

>   message: "Once the user reaches the character limit, they can’t enter more characters."

Reviewer: Nit-pick, but this just sounds better to my ear.
```suggestion
  message: "Once the user reaches the character limit, they can’t enter any more characters."
```

} %}

{% contentLayout %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/character-counter-empty.svg",
      width: "256px",

>       alt: "An empty textarea component with a label and character counter on the right top. Counter shows that the is zero from fifty symbols."

Reviewer 2: ```suggestion
      alt: "An empty textarea component with a label and character counter on the right top. Counter shows that there is zero from fifty symbols."
```

>       alt: "An empty textarea component with a label and character counter on the right top. Counter shows that there is zero from fifty symbols."

Reviewer: ```suggestion
      alt: "An empty textarea component with a label and character counter on the top right. The counter shows zero out of fifty characters."
```

    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/character-counter.svg",
      width: "256px",

>       alt: "Textarea component with a label and text inside it. Textarea's character counter on the right top of the component shows that there are fifty from fifty symbols."

Reviewer: ```suggestion
      alt: "A textarea component with a label, and text inside it. The character counter on the top right of the component shows fifty out of fifty characters."
```

    } %}
  {% endcontentItem %}
{% endcontentLayout %}

---

## Mobile devices

### Manual-resize

The manual-resize feature on mobile devices doesn't provide an optimal user experience. Therefore, it's replaced with a fixed height that enables users to scroll vertically.

{% contentLayout %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/fixed-height.svg",
      width: "256px",

>       alt: "Textarea component with a label and text inside it. Textarea's character counter on the right top of the component shows that there are fifty from fifty symbols."

Reviewer: ```suggestion
      alt: "A textarea component with a label, and text inside it. The character counter on the top right of the component shows fifty out of fifty characters."
```

    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/scrolled.svg",
      width: "256px",

>       alt: "An example of a textarea component with a lot of text content inside on a mobile device. This textarea has a fixed height and scroll functionality."

Reviewer: ```suggestion
      alt: "An example of a textarea component containing lots of text content on a mobile device. This textarea has a fixed height and scroll functionality."
```

      width: "256px",
      alt: "Textarea component with a label. Textarea is greyed out and has a black border to specify a read only state."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Error</h4>
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/error.svg",
      width: "256px",
      alt: "Textarea component with a label and red assistive text underneath specifying error state."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Error - hover</h4>
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/error-hover.svg",
      width: "256px",
      alt: "Textarea component with a label, red border and red assistive text underneath specifying error-hover state."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Error - focus</h4>
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/error-focus.svg",
      width: "256px",
      alt: "Textarea component with a label has focus, red border and red assistive text underneath specifying error-focus state."
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

---

## Examples

### LTR examples

> Here are some examples of Textareas in left-to-right context:

Reviewer: ```suggestion
Here are some examples of textareas in a left-to-right context:
```

{% contentLayout %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/example-ltr-auto.svg",
      width: "373px",
      alt: "An auto-resized textarea component with a label aligned to the left."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/textarea/example-ltr-manual.svg",
      width: "373px",
      alt: "A manual-resized textarea component with a label aligned to the left."
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

### RTL examples

> Here are some examples of Textareas in right-to-left context:

Reviewer: ```suggestion
Here are some examples of textareas in a right-to-left context:
```
